src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala

```diff
@@ -0,0 +1,223 @@
+  EServiceDescriptorState => CatalogManagementApiDescriptorState
+}
+import it.pagopa.pdnd.interop.uservice.partymanagement.client.model.{Organization => PartyManagementApiOrganization}
+import it.pagopa.pdnd.interop.uservice.purposemanagement.client.model.PurposeVersionState.{
+  ARCHIVED,
+  DRAFT,
+  WAITING_FOR_APPROVAL
+}
+import it.pagopa.pdnd.interop.uservice.purposemanagement.client.model.{
+  PurposeVersionState,
+  Purpose => PurposeManagementApiPurpose,
+  Purposes => PurposeManagementApiPurposes
+}
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+import java.time.OffsetDateTime
+import java.util.UUID
+import scala.annotation.nowarn
+import scala.util.{Failure, Success, Try}
+
+package object impl extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val problemErrorFormat: RootJsonFormat[ProblemError] = jsonFormat2(ProblemError)
+  implicit val problemFormat: RootJsonFormat[Problem]           = jsonFormat5(Problem)
+
+  implicit val clientCredentialsResponseFormat: RootJsonFormat[ClientCredentialsResponse] = jsonFormat3(
+    ClientCredentialsResponse
+  )
+
+  implicit val purposeFormat: RootJsonFormat[Purpose]   = jsonFormat3(Purpose)
+  implicit val purposesFormat: RootJsonFormat[Purposes] = jsonFormat1(Purposes)
+
+  implicit val subscriberFormat: RootJsonFormat[Organization] = jsonFormat3(Organization)
+  implicit val eServiceFormat: RootJsonFormat[EService]       = jsonFormat4(EService)
+
+  implicit val agreementFormat: RootJsonFormat[Agreement]   = jsonFormat6(Agreement)
+  implicit val agreementsFormat: RootJsonFormat[Agreements] = jsonFormat1(Agreements)
+
+  implicit val attributeFormat: RootJsonFormat[Attribute] = jsonFormat3(Attribute)
+
+  implicit val attributeValidityStateFormat: RootJsonFormat[AttributeValidityState] = jsonFormat2(
+    AttributeValidityState
+  )
+
+  implicit val attributesFormat: RootJsonFormat[Attributes] = jsonFormat1(Attributes)
+
+  final val serviceErrorCodePrefix: String = "013"
+  final val defaultProblemType: String     = "about:blank"
+
+  def problemOf(httpError: StatusCode, error: ComponentError, defaultMessage: String = "Unknown error"): Problem =
+    Problem(
+      `type` = defaultProblemType,
+      status = httpError.intValue,
+      title = httpError.defaultMessage,
+      errors = Seq(
+        ProblemError(
+          code = s"$serviceErrorCodePrefix-${error.code}",
+          detail = Option(error.getMessage).getOrElse(defaultMessage)
+        )
+      )
+    )
+
+  private val descendingOffsetDateTimeOrdering: Ordering[OffsetDateTime] = implicitly[Ordering[OffsetDateTime]].reverse
+
+  implicit class EnrichedPurpose(private val purpose: PurposeManagementApiPurpose) extends AnyVal {
+    def toModel: Try[Purpose] =
+      purpose.versions
+        .sortBy(_.createdAt)(descendingOffsetDateTimeOrdering)
+        .find(p => p.state == PurposeVersionState.ACTIVE && p.state == PurposeVersionState.SUSPENDED)
```

Comment:
why sorting if we find just one value?

Reply:
Cannot we have both a suspended and an active purpose version in the Seq? If so, shouldn't we return the most recent one?

Reply:
given a Purpose, only one version in Active or Suspended can exist. 
There can not be a version in Active and one in Suspended

Reply:
ok then, so do you suggest to modify the client first? Right?

Reply:
ok then. I am going to fix also the wrong boolean condition here.